pbft/pbft.cpp

```diff
@@ -907,33 +750,34 @@ pbft::do_committed(const std::shared_ptr<pbft_operation>& op)
     }
 }
 
-void
-pbft::handle_new_config_timeout(const boost::system::error_code& ec)
+bool
+pbft::is_primary() const
 {
-    if (ec == boost::asio::error::operation_aborted)
-    {
-        return;
-    }
-
-    if (ec)
-    {
-        LOG(error) << "handle_new_config_timeout error: " << ec.message();
-        return;
-    }
-
-    this->initiate_viewchange();
+    auto primary = this->get_current_primary();
+    return primary.has_value() && primary.value().uuid == this->uuid;
```

Comment:
Not that we "support" Mac, but .value() seems to cause a problem with Mac. Possibly use *primary instead.